File: java/fakeapply/src/main/java/io/pivotal/fakeapply/LoanApplication.java
package io.pivotal.fakeapply;



public class LoanApplication {

  private String uuid, name;
  private long amount;

  public LoanApplication() {
  }

  public LoanApplication(String uuid, String name, long amount) {
    this.uuid = uuid;
    this.name = name;
    this.amount = amount;
  }

  public String getUuid() {
    return uuid;
  }

  public String getName() {
    return name;
  }

  public long getAmount() {
    return amount;
  }

  @Override
  public String toString() {
    return "LoanApplication{" +
            "uuid='" + uuid + '\'' +
            ", name='" + name + '\'' +
            ", amount=" + amount +
            '}';
  }
}

File: loan-application-service/src/MyCompany.LoanApplication/Models/LoanApplication.cs
using System;
using Newtonsoft.Json;

namespace LoanApplication.Models
{
	public class LoanApplication
	{
		public string id { get; set; }
		public string name { get; set; }
		public double amount { get; set; }
		public string status { get; set; }

		public override string ToString()
		{
			return $"Loan[{this.id},{this.name},{this.amount},{this.status}]";
		}
		public string AsJson()
		{
			return JsonConvert.SerializeObject(this);
		}

		public LoanApplicationEntity AsLoanApplicationEntity(){
			if (string.IsNullOrEmpty(this.id))
				throw new NullReferenceException("Id");

			if (string.IsNullOrEmpty(this.status))
				throw new NullReferenceException("status");

			if (string.IsNullOrEmpty(this.name))
				throw new NullReferenceException("name");

			return new LoanApplicationEntity(){
				Id = Guid.Parse(this.id),
				FullName = this.name,
				Amount = this.amount,
				LoanStatus = (LoanStatus)Enum.Parse(typeof(LoanStatus), this.status)
			};
		}
	}
}
